exports/dataset.cpp

```diff
@@ -518,12 +532,12 @@ PYBIND11_MODULE(dataset, m) {
       .def(py::self += py::self, py::call_guard<py::gil_scoped_release>())
       .def(py::self -= py::self, py::call_guard<py::gil_scoped_release>())
       .def(py::self *= py::self, py::call_guard<py::gil_scoped_release>())
-      .def("__iadd__", [](VariableSlice &a, Variable &b) { return a += b; },
-           py::is_operator())
-      .def("__isub__", [](VariableSlice &a, Variable &b) { return a -= b; },
-           py::is_operator())
-      .def("__imul__", [](VariableSlice &a, Variable &b) { return a *= b; },
-           py::is_operator())
+      .def(py::self /= py::self, py::call_guard<py::gil_scoped_release>())
+      .def(py::self + py::self, py::call_guard<py::gil_scoped_release>())
+      .def(py::self - py::self, py::call_guard<py::gil_scoped_release>())
+      .def(py::self * py::self, py::call_guard<py::gil_scoped_release>())
+      .def(py::self / py::self, py::call_guard<py::gil_scoped_release>())
```

Comment:
Not sure why this was changed?
- The non-assigning operators (+-*/) can simply use implicit conversion to `Variable`, I think, and are thus not needed for `VariableSlice` for the first argument.
- You removed `+=`, `*=` and `-=`? Admittedly the old implementation was slightly inefficient since it did an implicit conversion to `Variable` in case the second operand was a `VariableSlice`, i.e., a second overload should probably be added.
- The added operators do not work for `Variable` as a second operand?